edb/server/compiler/compiler.py

```diff
@@ -526,17 +526,28 @@ def parse_search_path(search_path_str: str) -> list[str]:
                 if isinstance(arg, pgast.StringConstant)
             ]
 
-        frontend_only = {'search_path'}
+        # frontend-only settings and their mutability
+        frontend_only = {
+            'search_path': True,
+            'server_version': False,
+            'server_version_num': False,
+        }
         stmts = pg_parser.parse(query_str)
         sql_units = []
         for stmt in stmts:
+            fe_only = stmt.name in frontend_only
```

Comment:
Did you mean to write `fe_only = frontend_only[stmt.name]`?

Reply:
Ah no, that's a bad naming - let me update, thanks!